internal/controller/k0smotron.io/k0smotroncluster_service_test.go

```diff
@@ -127,3 +129,67 @@ func TestClusterReconciler_findNodeAddress(t *testing.T) {
 		})
 	}
 }
+
+func TestClusterReconciler_serviceAnnotations(t *testing.T) {
+	tests := []struct {
+		name string
+		kmc  *km.Cluster
+		want map[string]string
+	}{
+		{
+			name: "when no annotations are set on either Cluster on svc",
+			kmc: &km.Cluster{
+				ObjectMeta: metav1.ObjectMeta{
+					Name: "test",
+				},
+				Spec: km.ClusterSpec{},
+			},
+			want: map[string]string{},
+		},
+		{
+			name: "when annotations are set on only Cluster",
+			kmc: &km.Cluster{
+				ObjectMeta: metav1.ObjectMeta{
+					Name: "test",
+					Annotations: map[string]string{
+						"test": "test",
+					},
+				},
+				Spec: km.ClusterSpec{},
+			},
+			want: map[string]string{
+				"test": "test",
+			},
+		},
+		{
+			name: "when annotations are set on both Cluster and svc",
+			kmc: &km.Cluster{
+				ObjectMeta: metav1.ObjectMeta{
+					Name: "test",
+					Annotations: map[string]string{
+						"test": "test",
```

Comment:
I think it would make sense to test if an annotation is defined on both it gets the service annotation instead.
This small change should be enough:
```suggestion
						"test": "test",
						"foo":  "test",
```

I don't feel very strongly about this, so if you think it doesn't make sense then don't add it.

Reply:
TEsting this makes sense. I'll add a separate case for this